src/CoordinateSystemsCore/CoordinateSystemsCore.jl

```diff
@@ -0,0 +1,33 @@
+module CoordinateSystemsCore
+
+using StaticArrays
+
+export AbstractCoordinateSystem, Cartesian, Spherical
+export point, coordinate, coordinate_system, with_coordinate_system
+export AbstractPoint, Point
```

Comment:
I would remove the Abstract prefix from most types to make them short. Also, maybe we could leave the Point type outside this package and only deal with the coordinate transformations. The point itself can be defined in separate packages that consume coordinates.